packages/core/src/RenderingEngine/BaseVolumeViewport.ts

```diff
@@ -827,6 +830,61 @@ abstract class BaseVolumeViewport extends Viewport implements IVolumeViewport {
     };
   };
 
+
+  /**
+   *
+   * @param applicableVolumeActorInfo  - The volume actor information for the volume
+   * @returns colormap information for the volume if identified
+   */
+  private getColormap = (applicableVolumeActorInfo) => {
+    const { volumeActor } = applicableVolumeActorInfo;
+    const cfun = volumeActor.getProperty().getRGBTransferFunction(0);
+    const { nodes } = cfun.getState()
+    const RGBPoints = nodes.reduce((acc, node) => {
+      acc.push(node.x, node.r, node.g, node.b);
+      return acc;
+    }
+    , []);
+    const colormaps = vtkColorMaps.rgbPresetNames.map((presetName) =>
+    vtkColorMaps.getPresetByName(presetName)
+    );
+    const matchedColormap = colormaps.find((colormap) => {
+        const { RGBPoints: presetRGBPoints } = colormap;
+        if (presetRGBPoints.length !== RGBPoints.length) {
+            return false;
+        }
+        for (let i = 0; i < presetRGBPoints.length; i += 4) {
+            if (
+                presetRGBPoints[i + 1] !== RGBPoints[i + 1] ||
+                presetRGBPoints[i + 2] !== RGBPoints[i + 2] ||
+                presetRGBPoints[i + 3] !== RGBPoints[i + 3]
+            ) {
+                return false;
+            }
+        }
+
+        return true;
+    });
```

Comment:
you should be Epsilon compare (you can use isEqual which can run on vectors too)